providers/oidc_test.go

```diff
@@ -316,6 +322,53 @@ func TestCreateSessionStateFromBearerToken(t *testing.T) {
 	}
 }
 
+func TestCreateSessionStateFromBearerTokenGroups(t *testing.T) {
```

Comment:
There's a lot of duplicated code here with `TestCreateSessionStateFromBearerToken` 

Are you able to consolidate these two test cases and add these two test cases as `group` variants to the other test function?

Reply:
Collapsed the tests, I originally was avoiding complicating the existing test but in hindsight it isn't too bad, PTAL.

Reply:
@NickMeves let me know if anything else is needed to get this wrapped up, I am at a point where I would love to get this in so I can answer our AuthZ story on our side and finally close things off!